Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -456,7 +456,47 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
         ],
         handledNodes: [argument.id]
       )
-      return .visitChildren
+    }
+    if let unexpectedAtSign = node.unexpectedBeforeAtSignToken?.onlyToken(where: { $0.tokenKind == .atSign && !$0.trailingTrivia.isEmpty }) {
+      addDiagnostic(
+        unexpectedAtSign,
+        position: node.unexpectedBeforeAtSignToken?.endPosition.advanced(by: -1),
+        StaticParserError.invalidWhitespaceBetweenAttributeAtSignAndIdentifier,
+        fixIts: [
+          FixIt(
+            message: StaticParserFixIt.removeExtraneousWhitespace,
+            changes: [
+              .makeMissing(unexpectedAtSign, transferTrivia: false),
+              .makePresent(node.atSignToken),
+            ]
+          )
+        ],
+        handledNodes: [
+          node.id,
+          unexpectedAtSign.id,
+          node.atSignToken.id
+        ]
+      )
+    } else if let unexpectedIdentifier = node.unexpectedBetweenAtSignTokenAndAttributeName?.onlyToken(where: { $0.tokenKind.isIdentifier && !$0.leadingTrivia.isEmpty }),
+              node.attributeName.isMissingAllTokens {
```

Comment:
The `onlyToken(where: { $0.tokenKind.isIdentifier })` check fails when the type for the attribute has generic arguments, e.g. `@MyType<Int>`. I think all you need to check here is that `node.unexpectedBetweenAtSignTokenAndAttributeName?.node.leadingTriviaLength != 0` and then make the entire `node.unexpectedBetweenAtSignTokenAndAttributeName` missing, even if it contains multiple tokens.